python/cudf/cudf/core/index.py

```diff
@@ -2991,89 +2990,56 @@ def interval_range(
             closed='left',
             dtype='interval')
     """
-    if freq and periods and start and end:
+    nargs = sum(_ is not None for _ in (start, end, periods, freq))
+
+    # we need at least three of (start, end, periods, freq)
+    if nargs == 2 and freq is None:
+        freq = 1
+        nargs += 1
+
+    if nargs != 3:
         raise ValueError(
             "Of the four parameters: start, end, periods, and "
             "freq, exactly three must be specified"
         )
-    args = [
-        cudf.Scalar(x) if x is not None else None
-        for x in (start, end, freq, periods)
-    ]
+
+    start = cudf.Scalar(start) if start is not None else start
+    end = cudf.Scalar(end) if end is not None else end
+    periods = cudf.Scalar(int(periods)) if periods is not None else periods
+    freq = cudf.Scalar(freq) if freq is not None else freq
+
+    if start is None:
+        start = end - freq * periods
+    elif freq is None:
+        quotient, remainder = divmod((end - start).value, periods.value)
```

Comment:
`divmod` seems not to work with `cudf.Scalar`

Reply:
Can you do all this manipulation on the host before scalarising at the end, that would have far fewer syncs.

Reply:
We have tests written already that pass `Scalar` objects as input :(

Reply:
The scalar caching should save us here, no? There shouldn't be a sync until `device_value` is called later on. `.value` should just return the cached host value if the scalar was constructed from a host scalar